plugins/coreboot/fu-coreboot-fmap.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (C) 2019 9elements Agency GmbH <patrick.rudolph@9elements.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/mman.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-device-metadata.h"
+#include "fu-device-private.h"
+#include "fu-plugin-coreboot.h"
+#include "fmap_serialized.h"
+
+/* Depends on the kernel module coreboot-fmap.ko */
+
+struct fmap *
+fu_plugin_coreboot_find_fmap (FuPlugin *plugin, GError **error)
+{
+	gchar *tmp = NULL;
+
+	g_debug ("coreboot: Trying /sys/firmware/fmap");
+
+	if (!g_file_get_contents ("/sys/firmware/fmap", &tmp, NULL, error))
```

Comment:
I think that it makes sense to also check for existence of this sysfs file in startup.  If the system is running coreboot but this file is missing then you can set an error indicating lack of kernel support and avoid a bunch of plugin init.

Reply:
Not required as the FMAP only exposes partitions within coreboot. The coreboot device itself can still be exposed without the UPDATABLE flag set.